ext/mysqli/mysqli_nonapi.c

```diff
@@ -72,7 +72,12 @@ void mysqli_common_connect(INTERNAL_FUNCTION_PARAMETERS, bool is_real_connect, b
 	}
 #endif
 
-	if (getThis() && !ZEND_NUM_ARGS() && in_ctor) {
+	if (UNEXPECTED(in_ctor && (Z_MYSQLI_P(object))->ptr)) {
+		zend_throw_error(NULL, "Cannot call constructor twice");
+		return;
+	}
+
+	if (in_ctor && !ZEND_NUM_ARGS()) {
```

Comment:
I don't know why the code was checking for `getThis` before but I you are right. If we are in constructor then obviously we have this.